rotkehlchen/chain/ethereum/modules/adex/utils.py

```diff
@@ -145,7 +146,17 @@ def deserialize_adex_event_from_db(
             )
 
         try:
-            token = EthereumToken(event_tuple[13])   # type: ignore # type already checked
+            if event_tuple[13].startswith(ETHEREUM_DIRECTIVE):  # type: ignore # type checked
+                token = EthereumToken.from_identifier(
+                    event_tuple[13],   # type: ignore # type already checked
+                )
+                if token is None:
+                    raise DeserializationError(
+                        f'Unknown {event_tuple[13]} found while processing adex event. '
+                        f'Unexpected data: {event_tuple}',
+                    )
+            else:
+                token = EthereumToken(event_tuple[13])  # type: ignore # type already checked
```

Comment:
So this should work, the bug is indeed here.

But a few things need changing.

1. The if is not needed. As you can see from the schema in adex/typing.py this should always be a token. It's not any other asset so it should always start with _ceth_.
2. Try to remove the type: ignore. Not sure if it's needed. If it is just keep it.
3. The else can go away. The else would never work like this. The `EthereumToken()` constructor takes an address.

```suggestion
                token = EthereumToken.from_identifier(event_tuple[13])
                if token is None:
                    raise DeserializationError(
                        f'Unknown token {event_tuple[13]} found while processing adex event. '
                        f'Unexpected data: {event_tuple}',
                    )
```

Reply:
Yes I tried to remove the type ignore but is needed. Comment addressed